gammapy/spectrum/dataset.py

```diff
@@ -254,9 +254,13 @@ def fake(self, random_state="random-seed"):
     def energy_range(self):
         """Energy range defined by the safe mask"""
         energy = self._energy_axis.edges
-        e_lo = energy[:-1][self.mask_safe]
-        e_hi = energy[1:][self.mask_safe]
-        return u.Quantity([e_lo.min(), e_hi.max()])
+        e_min, e_max = energy[:-1], energy[1:]
+
+        if self.mask_safe is not None and self.mask_safe.any():
+            e_min = e_min[self.mask_safe]
```

Comment:
Can we add a `else` to return `None` if the mask is entirely `False`? I guess it will break the `IRFStacker`. We might add a test there to avoid calling it if one the the dataset has a `safe_mask` fully set to `None`

Reply:
I agree this is the better solution, it also addresses the comment by @lmohrmann.